src/compiler/core.ts

```diff
@@ -1438,12 +1438,18 @@ namespace ts {
         }
     }
 
-    export function group<T>(values: ReadonlyArray<T>, getGroupId: (value: T) => string): ReadonlyArray<ReadonlyArray<T>> {
-        const groupIdToGroup = createMultiMap<T>();
+    export function toMultiMap<T>(values: ReadonlyArray<T>, getKey: (value: T) => string): MultiMap<T>;
+    export function toMultiMap<T, U>(values: ReadonlyArray<T>, getKey: (value: T) => string, getValue: (value: T) => U): MultiMap<U>;
+    export function toMultiMap<T, U>(values: ReadonlyArray<T>, getKey: (value: T) => string, getValue: (value: T) => T | U = identity): MultiMap<T | U> {

```

Comment:
This overload has a slightly different signature than the corresponding one for `arrayToMap`.  Would it make sense to keep them in sync?